platooning_strategic_IHP/src/platoon_strategic_ihp.cpp

```diff
@@ -1376,6 +1916,17 @@ namespace platoon_strategic_ihp
                     // do not need to update platoon leader and platoon ID for candidate leader
 
                 }
+                // UCLA: CutIn join 
+                else if (isCutInJoin && msg.is_accepted)
+                {
+                    ROS_DEBUG_STREAM("Received positive response for plan id = " << pm_.current_plan.planId);
+                    ROS_DEBUG_STREAM("Change to Prepare to join state and prepare to change lane. ");
+
+                    // Change to candidate leader and idle
+                    pm_.current_platoon_state = PlatoonState::PREPARETOJOIN;
+                    candidatestateStartTime = ros::Time::now().toNSec() / 1000000;
+                    // do not need to update platoon leader and platoon ID for candidate leader
+                }
                 else
```

Comment:
This else block has been misinterpreted as only happening if the response was NACK (according to the debug stream).  This block will also get executed if host is the platoon leader, and (I think) all 3 boolean flags are false.